src/Compilers/CSharp/Portable/Binder/ForEachLoopBinder.cs

```diff
@@ -220,6 +220,11 @@ private BoundForEachStatement BindForEachPartsWorker(BindingDiagnosticBag diagno
                 CheckFeatureAvailability(_syntax.AwaitKeyword, MessageID.IDS_FeatureAsyncStreams, diagnostics);
             }
 
+            if (_syntax is ForEachStatementSyntax forEachStatement)
+            {
+                ReportFieldOrValueContextualKeywordConflictIfAny(forEachStatement, forEachStatement.Identifier, diagnostics);
```

Comment:
nit: have you considered placing this check in the switch below that already differentiates regular foreach from deconstruction foreach? (line 277)